src/Data/Function.purs

```diff
@@ -77,6 +81,16 @@ applyFlipped x f = f x
 -- | ```
 infixl 1 applyFlipped as #
 
+-- | Applies a function to an argument `n` times.
+-- |
+-- | If n is less than or equal to 0, the function is not applied.
+applyN :: forall a. (a -> a) -> Int -> a -> a
+applyN f n = go n
```

Comment:
I think eta-reducing this (`applyN f = go`) would be nice, so that there are fewer variables of type `Int` in scope, which means there's less to keep in your head while reading this. We then wouldn't need to use a primed identifier in the definition of `go`, I think.

Reply:
Also I think it might not be entirely clear what `n` is in the comments: how about this?

> The function `applyN f n` applies the function `f` to its argument `n` times.

I think examples using `f = (_+1)` are probably best, seeing as there are few other functions and data types in scope here for us to use. Examples using that `f` would hopefully be easy to understand, too.